decidim/decidim
# Use comments counter cache instead of additional query

#### :tophat: What? Why?
When browsing e.g. the blogs index, you can see from the benchmarks that there are bunch of `COUNT` queries done against the comments table.

The counter caches were already added at #6438 but they are not properly used.

It also does not work if we would add `counter_cache: :comments_count` to the commentable concern:
https://github.com/decidim/decidim/blob/96c78cbe478dbb82a44fa9f96275c832e76a8485/decidim-comments/lib/decidim/comments/commentable.rb#L13

Even after it is added there and the `Comment` model (as with other counter caches), the queries won't disappear.

The only way how I could figure out to get rid of the extra queries was by accessing the cache column directly.

#### :pushpin: Related Issues
- Related to #6438, #7574

#### Testing
- Create an instance with the test seeds
- Go to one of the blogs components
- Check the benchmark section at the top right
- See a bunch of `COUNT` queries against the comments table (see screenshot)

#### :clipboard: Checklist

### :camera: Screenshots
![Count queries in the benchmark tool](https://user-images.githubusercontent.com/864340/111368627-cac82080-869e-11eb-8293-7520d45ff72b.png)
